docs/src/pages/docs/crocks/Async.md

```diff
@@ -0,0 +1,1357 @@
+  safe(isString)
+)
+
+fake('file-a')
+  .fork(log('rej'), log('res'))
+//=> file-b
+
+// getTwo :: a -> Async String String
+const getTwo =
+  composeK(fake, fake)
+
+getTwo('file-a')
+  .fork(log('rej'), log('res'))
+//=> res: file-c
+
+getTwo('file-b')
+  .fork(log('rej'), log('res'))
+//=> rej: "Not Found"
+
+getTwo(76)
+  .fork(log('rej'), log('res'))
+//=> rej: "Not Safe"
+```
+
+#### coalesce
+
+```haskell
+Async e a ~> ((e -> b), (a -> b))) -> Async e b
+```
+Used as a means to apply a computation to a [`Resolved`](#resolved) instance and then map
+any [`Rejected`](#rejected) value while transforming it to a [`Resolved`](#resolved) to continue
+computation. `coalesce` on an `Async` can be used to model the all too
+familiar, and more imperative `if/else` flow in a more declarative manner.
+
+The first function is used when invoked on a [`Rejected`](#rejected) instance and will
+return a [`Resolved`](#resolved) instance wrapping the result of the function. The second
+function is used when `coalesce` is invoked on a [`Resolved`](#resolved) instance and is used
+to map the original value, returning a new [`Resolved`](#resolved) instance wrapping the
+result of the second function.
+
+```javascript
+const Async = require('crocks/Async')
+const { Rejected, Resolved } = Async
+
+const coalesce = require('crocks/pointfree/coalesce')
+
+// log :: String -> a -> a
+const log = label => x =>
+  (console.log(`${label}:`, x), x)
+
+// label :: String -> String -> String
+const label =
+  lbl => x => `${lbl} ${x}`
+
+// resolve :: Async e String -> Async e String
+const resolve =
+  coalesce(label('Was'), label('Still'))
+
+resolve(Resolved('Resolved'))
+  .fork(log('rej'), log('res'))
+//=> res: "Still Resolved"
+
+resolve(Rejected('Rejected'))
+  .fork(log('rej'), log('res'))
+//=> res: "Was Rejected"
+```
+
+#### swap
+
+```haskell
+Async e a ~> ((e -> b), (a -> c)) -> Async c b
+```
+
+Used to map the value of a [`Rejected`](#rejected) into a [`Resolved`](#resolved) or a [`Resolved`](#resolved) to
+a [`Rejected`](#rejected), `swap` takes (2) functions as its arguments. The first function
+is used to map the expected [`Rejected`](#rejected) value into a [`Resolved`](#resolved), while the
+second goes from [`Resolved`](#resolved) to [`Rejected`](#rejected). If no mapping is required on either,
+then [`identity`][identity] functions can be used in one or both arguments.
+
+```javascript
+const Async = require('crocks/Async')
+const { Rejected, Resolved } = Async
+
+const compose = require('crocks/helpers/compose')
+const identity = require('crocks/combinators/identity')
+const swap = require('crocks/pointfree/swap')
+
+// log :: String -> a -> a
+const log = label => x =>
+  (console.log(`${label}:`, x), x)
+
+// length :: String -> Integer
+const length =
+  x => x.length
+
+// repeat :: String -> Number -> String
+const repeat =
+  char => n => char.repeat(n)
+
+// values :: Async String Number -> Async String Number
```

Comment:
I'm still pretty new to reading signatures, but this doesn't feel right? Did you mean `values :: Async String Number -> Async Number String`?

Reply:
Nope, that siggy is correct as is. In this case we are map values to their respective types:

```js
// repeat :: String -> Number -> String
// length :: String -> Integer
```

so the `Rejected` which runs through `length`, will take the `String` and deposit a `Number` in the `Resolved` and the `repeat('a')` will take a `Resolved` `Number` and map it to a `String` before dumping it into a `Rejected`.

Using the functions to allow mapping keep from the `.swap().bimap(repeat('a'), length)` which for me at least occurs more often then just a straight `.swap()`. So the `crocks` `swaps` allow for mapping values.

If that makes sense?

Reply:
Sure, thanks for the clarification :-)